src/harness/testcases/WINNF_FT_S_WDB_testcase.py

```diff
@@ -0,0 +1,217 @@
+                                             device_b['userId'])
+
+    # Getting PAL IDs from pal records
+    pal_ids_a = [pal_records_a[0]['palId'], pal_records_a[1]['palId']]
+    pal_ids_b = [pal_records_b[0]['palId']]
+
+    # Creating conditionals for Cat B devices
+    self.assertEqual(device_b['cbsdCategory'], 'B')
+    conditionals_b = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['installationParam']
+    del device_b['measCapability']
+
+    conditionals = [conditionals_b]
+
+    # Create the actual config file
+    config = {
+        'palIds': [pal_ids_a, pal_ids_b],
+        'registrationRequests': [device_a, device_b],
+        'conditionalRegistrationData': conditionals
+        # TODO
+        # Need to add data base configurations.
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_WDB_1_default_config)
+  def test_WINNF_FT_S_WDB_1(self, config_filename):
+    """PAL Database Update."""
+
+    # Load the configuration file
+    config = loadConfig(config_filename)
+
+    # Very light checking of the config file.
+    self.assertEqual(len(config['registrationRequests']), len(config['palIds']))
+
+    # Step 1: Register device(s) 'C' with conditional registration data
+    cbsd_ids = self.assertRegistered(config['registrationRequests'],
+                                     config['conditionalRegistrationData'])
+
+    # Step 2: Request creation of PPA(s).
+    # Forming of PPA creation request(s)
+    ppa_creation_requests = []
+    for cbsd_id, pal_ids in zip(cbsd_ids, config['palIds']):
+      ppa_creation_requests.append({
+        "cbsdIds": [cbsd_id],
+        "palIds": pal_ids
+      })
+
+    # Trigger PPA creation
+    for ppa_creation_request in ppa_creation_requests:
+      self.assertPpaCreationFailure(ppa_creation_request)
+
+    # TODO
+    # Step 3: Create PAL database with a PAL record containg the PAL ID 'P'.
+
+    # Step 4: Admin Test Harness triggers CPAS
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Step 5: Request creation of PPA(s) again
+    for ppa_creation_request in ppa_creation_requests:
+      ppa_id = self.triggerPpaCreationAndWaitUntilComplete(ppa_creation_request)
+      logging.debug('ppa_id received from SAS UUT:%s', ppa_id)
+
+  def generate_WDB_2_default_config(self, filename):
+    """Generates the WinnForum configuration for WDB.2"""
+
+    # Load category B devices info
+    device_b_0 = json.load(
+         open(os.path.join('testcases', 'testdata', 'device_b.json')))
+    device_b_1 = json.load(
```

Comment:
Since you have not changed the FCC ID/SN, these look like the same device to the SAS UUT. You should just load another Cat B device instead of reusing device_b.json.

Reply:
used   "device_d" as another cat B device